state-chain/runtime/src/chainflip.rs

```diff
@@ -638,3 +639,13 @@ pub fn calculate_account_apy(account_id: &AccountId) -> Option<u32> {
 			.unwrap_or_default()
 	})
 }
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, Eq, Debug, Encode, Decode)]
+pub struct BlockUpdate<Data> {
+	pub block_hash: Hash,
+	pub block_number: BlockNumber,
+	// NOTE: Flatten requires Data types to be struct or map
+	// Also flatten is incompatible with u128, so AssetAmounts needs to be String type.
+	#[serde(flatten)]
+	pub data: Data,
```

Comment:
Do we really *need* to flatten here, given that it imposes these slightly weird conditions (AssetAmount is a string..)? 

Ie. could it not just be `update: <object>`?

Reply:
Its really about, how I want the JSON to look, if we have that problem happy to just duplicate the structures in the future.